<launch>
    <arg name="map_frame" default="map"/>
    <arg name="robot_frame" default="base_footprint"/>
    <arg name="scan_topic" default="/scan"/>
    <arg name="map_topic" default="/map"/>
    <arg name="sensor_data_topic" default="/sensor_data"/>
    
    <node name="sensor_interface" pkg="turtlebot3_explorer" type="sensor_interface" output="screen">
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="robot_frame" value="$(arg robot_frame)"/>
        <remap from="/scan" to="$(arg scan_topic)"/>
        <remap from="/map" to="$(arg map_topic)"/>
        <remap from="/sensor_data" to="$(arg sensor_data_topic)"/>
    </node>
</launch>